pcdet/models/detectors/pv_rcnn_ssl.py

```diff
@@ -341,15 +341,18 @@ def calc_statistics(self, pred_boxes, gt_boxes, update_metrics=True, tag=None, *
 
                 fg_thresh = self.model_cfg['ROI_HEAD']['TARGET_CONFIG']['CLS_FG_THRESH']
                 bg_thresh = self.model_cfg['ROI_HEAD']['TARGET_CONFIG']['CLS_BG_THRESH']
-                fg = (preds_iou_max > fg_thresh).float().mean().item()
-                statistics['pseudo_fgs'].append(fg)
+                filtered_fg = (preds_iou_max > fg_thresh).float()
+                filtered_bg = (preds_iou_max < bg_thresh).float()
+
+                statistics['pseudo_fgs'].append(filtered_fg.mean().item())
+                statistics['pseudo_bgs'].append(filtered_bg.mean().item())
 
                 if kwargs['pseudo_sem_scores']:
                     pseudo_sem_scores = kwargs['pseudo_sem_scores']
-                    sem_score_fg = (pseudo_sem_scores[i] * (preds_iou_max > fg_thresh).float()).sum() \
-                                   / torch.clamp((preds_iou_max > fg_thresh).sum(), min=1.0)
-                    sem_score_bg = (pseudo_sem_scores[i] * (preds_iou_max < bg_thresh).float()).sum() \
-                                   / torch.clamp((preds_iou_max < bg_thresh).float().sum(), min=1.0)
+                    sem_score_fg = (pseudo_sem_scores[i][torch.nonzero(valid_preds_mask)] * filtered_fg).sum() \
```

Comment:
And it seems due to [incorrect valid mask](https://github.com/fnozarian/OpenPCDet/pull/10/files#r966239808), the `pseudo_sem_scores` does not match with the `filtered_fg` (which should match in general and there's no need to use `valid_pred_mask` for `pseudo_sem_scores`).